src/nnet3/nnet-chain-training.cc

```diff
@@ -70,21 +70,40 @@ void NnetChainTrainer::Train(const NnetChainExample &chain_eg) {
 
   NnetComputer computer(nnet_config.compute_config, *computation,
```

Comment:
best to declare this object later on, just before it's needed.  otherwise it's confusing.